---
layout: page
permalink: /publications/
title: Publications
description: Peer-reviewed papers and preprints with links to code and data
---

- My pages on
  [Google Scholar](https://scholar.google.ca/citations?user=c83d8tgAAAAJ)
  and
  [Mathematics Genealogy Project](https://www.genealogy.math.ndsu.nodak.edu/id.php?id=224469).
- Peer review notice: in addition to peer-reviewed journals, I publish
  papers at machine learning conferences such as ICML and NIPS, which
  have double-blind peer reviews, and only accept about 20% of
  submitted papers.
- Reproducible Research notice: a "Reproducible" link for each paper below
  provides source code and data that I used to perform the analysis and make
  the figures/tables.
- Software notice: a "Software" link for each paper below provides a software
  package with a reference implementation of the algorithms described
  in our paper.
- Selected publications in [bib](/assets/TDH-refs.bib) format.

### In progress

- Hocking TD. mlr3resampling: an R implementation of cross-validation
  for comparing models learned using different train sets.
- Agyapong D, Propster JR, Marks J, Hocking TD. Cross-Validation for
  Training and Testing Co-occurrence Network Inference
  Algorithms. [Preprint arXiv:2309.15225](https://arxiv.org/abs/2309.15225).
- Why does functional pruning yield such fast algorithms for optimal
  changepoint detection? Invited talk for TRIPODS seminar
  [video](https://arizona.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=4e87c8d0-96d2-40d1-808c-ad16014c6962),
  [slides](https://github.com/tdhock/functional-pruning-theory#readme),
  [IEEE NJACS](https://events.vtools.ieee.org/m/280515), [ASU West ML
  Day](https://newcollege.asu.edu/machine-learning-day-1).
- Runge V, Hocking TD, Romano G, Afghah F, Fearnhead P, Rigaill
  G. gfpop: an R Package for Univariate Graph-Constrained Change-point
  Detection. Under review at Journal of Statistical
  Software, [Preprint
  arXiv:2002.03646](https://arxiv.org/abs/2002.03646),
  [Software (gfpop)](https://github.com/vrunge/gfpop),
  [Software (gfpopgui)](https://github.com/julianstanley/gfpopgui),
  [Reproducible](https://github.com/vrunge/gfpop/blob/master/vignettes/applications.Rmd).
- Venuto D, Hocking TD, Spanurattana S, Sugiyama M. Support vector
  comparison machines. [Preprint
  arXiv:1401.8008](https://arxiv.org/abs/1401.8008). Under review at
  Machine Learning,
  [Software](https://github.com/tdhock/rankSVMcompare),
  [Reproducible](https://github.com/tdhock/compare-paper).
- Hocking TD and Killick R. Changepoint detection algorithms and
  applications in R. Textbook in preparation.
- Hocking TD. A breakpoint detection error function for segmentation
  model selection and
  validation. [Preprint arXiv:1509.00368](https://arxiv.org/abs/1509.00368),
  [Software](http://r-forge.r-project.org/scm/?group_id=1540),
  [Reproducible](https://github.com/tdhock/breakpointError-orig).
- Hocking TD, Khare A. Learning penalty functions for changepoint
  detection using elastic-net regularized accelerated failure time
  models. In preparation, [Software](https://github.com/anujkhare/iregnet),
  [Reproducible](https://github.com/tdhock/iregnet-paper).
- Rust K, Hocking TD. A Log-linear Gradient Descent Algorithm for
  Unbalanced Binary Classification using the All Pairs Squared Hinge
  Loss. [Preprint arXiv:2302.11062](https://arxiv.org/abs/2302.11062).
- Hocking TD. Efficient line search optimization of penalty functions
  in supervised changepoint detection. [Talk announcement for JSM'23
  Toronto](https://ww2.aievolution.com/JSMAnnual/index.cfm?do=ev.viewEv&ev=2810),
  [Video of talk at Université Laval July
  2023](https://www.youtube.com/watch?v=22ODpDTZ4VE), [Slides
  PDF](https://github.com/tdhock/max-generalized-auc/blob/master/HOCKING-slides-toronto.pdf),
  [source](https://github.com/tdhock/max-generalized-auc/#4-may-2023).
- Hocking TD. Finite Sample Complexity Analysis of Binary Segmentation.
- Hocking TD. Comparing binsegRcpp with other implementations of
  binary segmentation for changepoint detection.
- Hocking TD. Teaching Hidden Markov Models Using Interactive Data
  Visualization. 
- Gabi Thibault, TD Hocking, Alexis Achim. Predicting forest burn from
  satellite image data.
- Feng Tao, Johannes Lehmann, Ying-Ping Wang, Lifen Jiang, Bernhard
  Ahrens, Kostiantyn Viatkin, Stefano Manzoni, Benjamin Z. Houlton,
  Yuanyuan Huang, Bruce A. Hungate, Serita D. Frey, Michael Schmidt,
  Markus Reichstein, Nuno Carvalhais, Philippe Ciais, Umakant Mishra,
  Gustaf Hugelius, Toby D. Hocking, Xingjie Lu, Zheng Shi, Ronald
  Vargas, Yusuf Yigini, Christian Omuto, Ashish A. Malik, Guillermo
  Peralta, Rosa Cuevas-Corona, Luciano E. Di Paolo, Isabel Luotto,
  Cuijuan Liao, Yi-Shuang Liang, Vinisa S. Saynes, Xiaomeng Huang,
  Yiqi Luo. Reply to: Beyond microbial carbon use efficiency. Preprint
  [EarthArXiv:6125](https://eartharxiv.org/repository/view/6125/).
 
### Accepted, in press

### 2024

- Kaufman J, Stenberg A, Hocking TD. Functional Labeled Optimal
  Partitioning. Published online 05 Jan 2024 by Journal of
  Computational and Graphical Statistics,
  [DOI:10.1080/10618600.2023.2293216](https://doi.org/10.1080/10618600.2023.2293216). [Preprint
  arXiv:2210.02580](https://arxiv.org/abs/2210.02580).
  
### 2023

- Harshe K, Williams J, Hocking TD, Lerner Z. Predicting Neuromuscular
  Engagement to Improve Gait Training with a Robotic Ankle
  Exoskeleton. [IEEE Robotics and Automation
  Letters](https://ieeexplore.ieee.org/document/10172008), vol. 8,
  no. 8, pp. 5055-5060, Aug. 2023, doi: 10.1109/LRA.2023.3291919.
- Tao F, Huang Y, Hungate BA, Manzoni S, Frey SD, Schmidt MWI,
  Reichstein M, Carvalhais N, Ciais P, Jiang L, Lehmann J, Mishra U,
  Hugelius G, Hocking TD, Lu X, Shi Z, Viatkin K, Vargas R, Yigini Y,
  Omuto C, Malik AA, Peralta G, Cuevas-Corona R, Di Paolo LE, Luotto
  I, Liao C, Liang YS, Saynes VS, Huang X, Luo Y. Microbial carbon use
  efficiency promotes global soil carbon
  storage. [Nature](https://www.nature.com/articles/s41586-023-06042-3)
  (2023),
  [DOI:10.1038/s41586-023-06042-3](https://doi.org/10.1038/s41586-023-06042-3).
- Hillman J, Hocking TD. Optimizing ROC Curves with a Sort-Based
  Surrogate Loss Function for Binary Classification and Changepoint
  Detection. [Journal of Machine Learning
  Research](https://jmlr.org/papers/v24/21-0751.html)
  24(70):1-24, 2023.  [Preprint
  arXiv:2107.01285](https://arxiv.org/abs/2107.01285),
  [Software](https://github.com/tdhock/aum),
  [Reproducible](https://github.com/tdhock/max-generalized-auc).
- Hocking TD, Srivastava A. Labeled Optimal Partitioning. Published in
  [Computational Statistics](https://rdcu.be/cQ8qM), (2023) 38:461-480.
  DOI:10.1007/s00180-022-01238-z, [Preprint
  arXiv:2006.13967](https://arxiv.org/abs/2006.13967),
  [Video](https://www.youtube.com/watch?v=lm_6_33zOWc),
  [Software](https://github.com/tdhock/LOPART),
  [Reproducible](https://github.com/tdhock/LOPART-paper).

### 2022

- Barr J, Hocking TD, Morton G, Thatcher T, Shaw P. Classifying
  Imbalanced Data with AUM Loss. 2022 Fourth International Conference
  on Transdisciplinary AI
  (TransAI). [IEEE](https://ieeexplore.ieee.org/document/9951523).
- Hocking TD, Barr J, Thatcher T. Interpretable linear models for
  predicting security vulnerabilities in source code. 2022 Fourth International Conference
  on Transdisciplinary AI
  (TransAI). [IEEE](https://ieeexplore.ieee.org/document/9951610/)
- Barr J, Shaw P, Abu-Khzam FN, Thatcher T, Hocking TD. Graph
  Embedding: A Methodological Survey. 2022 Fourth International Conference
  on Transdisciplinary AI
  (TransAI). [IEEE](https://ieeexplore.ieee.org/document/9951469/)
- Mihaljevic JR, Borkovec S, Ratnavale S, Hocking TD, Banister KE,
  Eppinger JE, Hepp CM, Doerry E. Rapid simulations of spatially
  explicit and stochastic models of infectious disease. [Preprint
  medRxiv](https://www.medrxiv.org/content/10.1101/2021.05.13.21256216v1). [Biology
  Methods &
  Protocols](https://academic.oup.com/biomethods/advance-article/doi/10.1093/biomethods/bpac022/6680179)
  2022, [Software](https://github.com/NAU-CCL/SPARSEMODr).
- Hocking TD. Chapter "Introduction to machine learning and neural
  networks" for book "Land Carbon Cycle Modeling: Matrix Approach,
  Data Assimilation, and Ecological Forecasting" edited by Luo Y,
  published by Taylor and Francis, [download whole book](https://www.taylorfrancis.com/books/oa-edit/10.1201/9780429155659/land-carbon-cycle-modeling-yiqi-luo-benjamin-smith),
  [download my chapter](https://raw.githubusercontent.com/tdhock/2020-yiqi-summer-school/master/HOCKING-chapter.pdf),
  [Reproducible](https://github.com/tdhock/2020-yiqi-summer-school#prepared-for-the-summer-school-4th-year-2021).
- Barnwal A, Cho H, Hocking TD. Survival regression with accelerated
  failure time model in XGBoost. [Journal of Computational and
  Graphical Statistics](https://www.tandfonline.com/doi/full/10.1080/10618600.2022.2067548) (doi:10.1080/10618600.2022.2067548). [Preprint
  arXiv:2006.04920](https://arxiv.org/abs/2006.04920),
  [Software](https://github.com/dmlc/xgboost),
  [Documentation](https://xgboost.readthedocs.io/en/latest/tutorials/aft_survival_analysis.html),
  [Video](https://www.youtube.com/watch?v=HuWRnzgGuIo),
  [Reproducible](https://github.com/avinashbarnwal/aftXgboostPaper/).
- Chaves AP, Egbert J, Hocking TD, Doerry E, Gerosa MA. Chatbots
  language design: the influence of language use on user
  experience. [ACM Transactions on Computer-Human
  Interaction](https://dl.acm.org/doi/10.1145/3487193) 29, 2, Article
  13 (2022). [Preprint
  arXiv:2101.11089](https://arxiv.org/abs/2101.11089).
- Hocking TD, Rigaill G, Fearnhead P, Bourque G. Generalized
  Functional Pruning Optimal Partitioning (GFPOP) for Constrained
  Changepoint Detection in Genomic Data.  [Journal of Statistical
  Software Vol. 101, Issue
  10](https://www.jstatsoft.org/article/view/v101i10). [Software](https://github.com/tdhock/PeakSegDisk),
  [Reproducible](https://github.com/tdhock/PeakSegFPOP-paper).
  Presentation at useR 2019:
  [slides](http://www.user2019.fr/static/pres/t257847.pdf),
  [video](https://www.youtube.com/watch?v=XlC4WCqsbuI).
  
### 2021

- Kolla A, Hocking TD, Groce A. Fuzz Testing the Compiled Code in R
  Packages. Published in proceedings of [ISSRE
  2021](https://ieeexplore.ieee.org/document/9700272).
- Hocking TD, Vargovich J. Linear Time Dynamic Programming for
  Computing Breakpoints in the Regularization Path of Models Selected
  From a Finite Set. [Journal of Computational and Graphical
  Statistics](https://amstat.tandfonline.com/doi/full/10.1080/10618600.2021.2000422)
  (2021), doi:10.1080/10618600.2021.2000422. [Preprint
  arXiv:2003.02808](https://arxiv.org/abs/2003.02808),
  [Software](https://github.com/tdhock/penaltyLearning),
  [Reproducible](https://github.com/tdhock/changepoint-data-structure#source-code-for-figures-in-paper).
- Kolla A, Hocking TD, Groce A. RcppDeepState, a simple way to fuzz
  test Rcpp Packages. Contributed talk at international useR 2021
  conference. [Software](https://github.com/akhikolla/RcppDeepState),
  [abstract](https://www.conftool.org/user2021/index.php?page=browseSessions&form_session=9#paperID123),
  [video](https://www.youtube.com/watch?v=LXyruwSo2K0&t=245s),
  [blog](https://akhikolla.github.io),
  [results](https://akhikolla.github.io/packages-folders/)
- Hocking TD. Wide-to-tall data reshaping using regular expressions
  and the nc package. [R
  Journal](https://journal.r-project.org/archive/2021/RJ-2021-029/index.html)
  (2021), doi:10.32614/RJ-2021-029, [Software](https://github.com/tdhock/nc),
  [Reproducible](https://github.com/tdhock/nc-article).
- Liehrmann A, Rigaill G, Hocking TD. Increased peak detection
  accuracy in over-dispersed ChIP-seq data with supervised
  segmentation models. [BMC Bioinformatics 22, Article number: 323
  (2021)](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04221-5),
  [Reproducible](https://github.com/aLiehrmann/ChIP_SEQ_segmentation_paper),
  [Software](https://github.com/aliehrmann/crocs).
- Fotoohinasab A, Hocking TD, Afghah F. A Greedy Graph Search
  Algorithm Based on Changepoint Analysis for Automatic QRS-Complex
  Detection. [Computers in Biology and Medicine 130 (2021)](https://www.sciencedirect.com/science/article/pii/S0010482521000020), [Preprint
  arXiv:2004.13558](https://arxiv.org/abs/2004.13558).
- Abraham A, Prys-Jones T, De Cuyper A, Ridenour C, Hempson G, Hocking
  TD, Clauss M, Doughty C. Improved estimation of gut passage time
  considerably affects trait-based dispersal models. 
  [Functional Ecology](https://besjournals.onlinelibrary.wiley.com/doi/10.1111/1365-2435.13726) 2021; 35: 860-869.
  
### 2020

- Fotoohinasab A, Hocking TD, Afghah F. A Graph-Constrained
  Changepoint Learning Approach for Automatic QRS-Complex
  Detection. [Paper](https://ieeexplore.ieee.org/document/9443307) in
  proceedings of [Asilomar Conference on Signals, Systems, and
  Computers](https://www2.securecms.com/Asilomar2020/Papers/ViewPapers.asp?PaperNum=1323)
  (2020), [Preprint
  arXiv:2004.13558](https://arxiv.org/abs/2004.13558).
- Fotoohinasab A, Hocking TD, Afghah F. A Graph-constrained
  Changepoint Detection Approach for ECG Segmentation. In proceedings
  of 42th Annual International Conference of the IEEE Engineering in
  Medicine and Biology Society (EMBC 2020), [IEEE
	  Xplore](https://ieeexplore.ieee.org/document/9175333).
- Hocking TD, Rigaill G, Fearnhead P, Bourque G. Constrained Dynamic
  Programming and Supervised Penalty Learning Algorithms for Peak
  Detection in Genomic Data. Journal of Machine Learning Research
  21(87):1-40, 2020. [JMLR](http://jmlr.org/papers/v21/18-843.html),
  [Preprint arXiv:1703.03352](https://arxiv.org/abs/1703.03352),
  [Software](https://github.com/tdhock/PeakSegOptimal),
  [Reproducible](https://github.com/tdhock/PeakSegFPOP-paper).
- Hocking TD, Bourque G. Machine Learning Algorithms for Simultaneous
  Supervised Detection of Peaks in Multiple Samples and Cell
  Types. Pacific Symposium on Biocomputing (2020)
  25:367-378, [PDF](http://psb.stanford.edu/psb-online/proceedings/psb20/Hocking.pdf),
  [Software](https://github.com/tdhock/PeakSegJoint).
  
### 2019

- Hocking TD. Comparing namedCapture with other R packages for regular
  expressions. 
  [R Journal](https://journal.r-project.org/archive/2019/RJ-2019-050/index.html) 
  (2019), doi:10.32614/RJ-2019-050 
  [Software](https://github.com/tdhock/namedCapture), 
  [Reproducible](https://github.com/tdhock/namedCapture-article).
- Jewell S, Hocking TD, Fearnhead P, Witten D. Fast Nonconvex
  Deconvolution of Calcium Imaging Data. Biostatistics (2019), doi:
  10.1093/biostatistics/kxy083,
  [pubmed](https://www.ncbi.nlm.nih.gov/pubmed/30753436).
  
### 2018

- Depuydt P, Koster J, Boeva V, Hocking TD, Speleman F, Schleiermacher
  G, De Preter K. Meta-mining of copy number profiles of high-risk
  neuroblastoma tumors. [Scientific Data](https://www.nature.com/articles/sdata2018240) (2018).
- Alirezaie N, Kernohan KD, Hartley T, Majewski J, Hocking
  TD. ClinPred: Prediction Tool to Identify Disease-Relevant
  Nonsynonymous Single-Nucleotide Variants. American
  Journal of Human Genetics
  (2018). [doi:10.1016/j.ajhg.2018.08.005](https://doi.org/10.1016/j.ajhg.2018.08.005)
  [AJHG](https://www.cell.com/action/showPdf?pii=S0002-9297%2818%2930271-4),
  [Software](http://github.com/tdhock/predict-clinically-pathogenic),
  [used in dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP).
- Sievert C, Cai J, VanderPlas S, Ferris K, Khan FUF, Hocking
  TD. Extending ggplot2 for linked and animated web graphics. Journal
  of Computational and Graphical Statistics (2018).
  doi:10.1080/10618600.2018.1513367.
  [JCGS](https://amstat.tandfonline.com/doi/full/10.1080/10618600.2018.1513367),
  [Software](https://github.com/tdhock/animint),
  [Reproducible](https://github.com/tdhock/animint-paper),
  [Interactive Figures](https://rcdata.nau.edu/genomic-ml/public_html/animint-paper-figures/).
- Depuydt P, Boeva V, Hocking TD, Cannoodt R, Ambros IM, Ambros PF,
  Asgharzadeh S, Attiyeh EF, Combaret V, Defferrari R, Fischer M, Hero
  B, Hogarty MD, Irwin MS, Koster J, Kreissman S, Ladenstein R,
  Lapouble E, Laureys G, London WB, Mazzocco K, Nakagawara A, Noguera
  R, Ohira M, Park JR, Pötschger U, Theissen J, Tonini GP,
  Valteau-Couanet D, Varesio L, Versteeg R, Speleman F, Maris JM,
  Schleiermacher G, De Preter K. Genomic Amplifications and Distal 6q
  Loss: Novel Markers for Poor Survival in High-risk Neuroblastoma
  Patients. Journal of the National Cancer
  Institute (2018), DOI:10.1093/jnci/djy022,
  [JNCI](https://academic.oup.com/jnci/advance-article/doi/10.1093/jnci/djy022/4921185).

### 2017

- Drouin A, Hocking TD, Laviolette F. Maximum margin interval
  trees. Neural Information Processing Systems
  (NeurIPS) 2017. [Software](https://github.com/aldro61/mmit),
  [Reproducible](https://github.com/tdhock/mmit-paper),
  [Preprint arXiv](https://arxiv.org/abs/1710.04234),
  [NeurIPS](http://papers.nips.cc/paper/7080-maximum-margin-interval-trees),
  [video](https://www.youtube.com/watch?v=sNrMH9z1rb4).
- Hocking TD, Goerner-Potvin P, Morin A, Shao X, Pastinen T, Bourque
  G. Optimizing ChIP-seq peak detectors using visual labels and
  supervised machine learning.  Bioinformatics (2017) 33 (4):
  491-499. [pubmed](https://www.ncbi.nlm.nih.gov/pubmed/27797775),
  [Software](https://github.com/tdhock/PeakError),
  [Reproducible](https://bitbucket.org/mugqic/chip-seq-paper),
  [Data](https://rcdata.nau.edu/genomic-ml/chip-seq-chunk-db/).
- Hocking TD and Killick R. Introduction to optimal changepoint
  detection algorithms. [useR2017 conference
  tutorial](https://www.user2017.brussels/uploads/TobyHocking.html),
  [video](https://channel9.msdn.com/Events/useR-international-R-User-conferences/useR-International-R-User-2017-Conference/Introduction-to-optimal-changepoint-detection-algorithms-II),
  [Reproducible](https://github.com/tdhock/change-tutorial).

### 2016

- Shimada K, Shimada S, Sugimoto K, Nakatochi M, Suguro M, Hirakawa A,
  Hocking TD, Takeuchi I, Tokunaga T, Takagi Y, Sakamoto A, Aoki T,
  Naoe T, Nakamura S, Hayakawa F, Seto M, Tomita A, Kiyoi
  H. Development and analysis of patient-derived xenograft mouse
  models in intravascular large B-cell lymphoma.
  Leukemia 2016. [pubmed](https://www.ncbi.nlm.nih.gov/pubmed/27001523).
- Chicard M, Boyault S, Colmet-Daage L, Richer W, Gentien D, Pierron
  G, Lapouble E, Bellini A, Clement N, Iacono I, Bréjon S, Carrere M,
  Reyes C, Hocking TD, Bernard V, Peuchmaur M, Corradini N,
  Faure-Conter C, Coze C, Plantaz D, Defachelles A-S, Thebaud E,
  Gambart M, Millot F, Valteau-Couanet D, Michon J, Puisieux A,
  Delattre O, Combaret V, Schleiermacher G. Genomic copy number
  profiling using circulating free tumor DNA highlights heterogeneity
  in neuroblastoma. Clinical Cancer
  Research 2016. [journal](http://clincancerres.aacrjournals.org/content/early/2016/11/03/1078-0432.CCR-16-0500).
- Maidstone R, Hocking TD, Rigaill G, Fearnhead P. On optimal multiple
  changepoint algorithms for large data. Statistics and Computing
  (2016). doi:10.1007/s11222-016-9636-3
  [journal](https://link.springer.com/article/10.1007/s11222-016-9636-3),
  [Software](https://r-forge.r-project.org/R/?group_id%3D1851),
  [Reproducible](https://r-forge.r-project.org/scm/viewvc.php/benchmark/?root%3Dopfp).
- The animint2 manual, [online
  "book" with multiple chapters](https://rcdata.nau.edu/genomic-ml/animint2-manual/Ch02-ggplot2.html),
  [Reproducible](https://github.com/tdhock/animint-book).
- Understanding and Creating Interactive Graphics. Hocking TD, Ekstrøm
  CT. [useR2016 conference
  tutorial](http://user2016.r-project.org/tutorials/13.html),
  [Reproducible](https://github.com/tdhock/interactive-tutorial).

### 2015

- Hocking TD, Rigaill G, Bourque G. PeakSeg: constrained optimal
  segmentation and supervised penalty learning for peak detection in
  count data.
  International Conference on Machine Learning (ICML), 2015. 
  [PMLR](http://proceedings.mlr.press/v37/hocking15.html), 
  [video](http://videolectures.net/icml2015_hocking_count_data/?q=hocking),
  [Software](https://github.com/tdhock/PeakSegDP),
  [Reproducible](https://github.com/tdhock/PeakSeg-paper).
- Hocking TD, Bourque G. PeakSegJoint: fast supervised peak detection
  via joint segmentation of multiple count data
  samples. [Preprint arXiv:1506.01286](https://arxiv.org/abs/1506.01286),
  [Software](https://github.com/tdhock/PeakSegJoint),
  [Reproducible](https://github.com/tdhock/PeakSegJoint-paper).
- Animint: Interactive Web-Based Animations Using Ggplot2's Grammar of
  Graphics. VanderPlas SR and Sievert C and Hocking TD. Presentation
  at
  [JSM2015](https://ww2.amstat.org/meetings/jsm/2015/onlineprogram/AbstractDetails.cfm?abstractid=314184%0A).

### 2014

- Suguro M, Yoshida N, Umino A, Kato H, Tagawa H, Nakagawa M, Fukuhara
  N, Karnan S, Takeuchi I, Hocking TD, Arita K, Karube K, Tsuzuki S,
  Nakamura S, Kinoshita T, Seto M. Clonal heterogeneity of lymphoid
  malignancies correlates with poor prognosis.  Cancer Sci. 2014
  Jul;105(7):897-904. [pubmed](http://www.ncbi.nlm.nih.gov/pubmed/24815991).
- Hocking TD, Boeva V, Rigaill G, Schleiermacher G, Janoueix-Lerosey
  I, Delattre O, Richer W, Bourdeaut F, Suguro M, Seto M, Bach F, Vert
  J-P. SegAnnDB: interactive Web-based genomic segmentation.
  Bioinformatics (2014) 30 (11):
  1539-1546. DOI:10.1093/bioinformatics/btu072
  [pubmed](http://www.ncbi.nlm.nih.gov/pubmed/24493034),
  [Software](https://github.com/tdhock/SegAnnDB),
  [Reproducible](https://gforge.inria.fr/scm/viewvc.php/breakpoints/webapp/applications-note/).

### 2013

- Hocking TD, Wutzler T, Ponting K and Grosjean P. Sustainable,
  extensible documentation generation using inlinedocs.  Journal of
  Statistical Software (2013), 54(6), 1-20. DOI:10.18637/jss.v054.i06
  [journal](https://www.jstatsoft.org/article/view/v054i06),
  [Software](https://cran.r-project.org/package=inlinedocs),
  [Reproducible](https://r-forge.r-project.org/scm/viewvc.php/tex/jss762/?root=inlinedocs).
- Hocking TD, Schleiermacher G, Janoueix-Lerosey I, Boeva V, Cappo J,
  Delattre O, Bach F, Vert J-P. Learning smoothing models of copy
  number profiles using breakpoint annotations.  BMC Bioinfo. 2013,
  14:164. DOI:10.1186/1471-2105-14-164
  [journal](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-164),
  [Software](https://cran.r-project.org/src/contrib/Archive/bams/),
  [Reproducible](https://rcdata.nau.edu/genomic-ml/public_html/papers/2012-01-27-Breakpoint-annotation-model-smoothing/HOCKING-breakpoint-annotation-model-smoothing.tgz).
- Hocking TD, Rigaill G, Bach F, Vert J-P. Learning sparse penalties
  for change-point detection using max-margin interval regression.
  International Conference on Machine Learning
  (ICML), 2013. [PMLR](http://proceedings.mlr.press/v28/hocking13.html),
  [video](http://techtalks.tv/talks/learning-sparse-penalties-for-change-point-detection-using-max-margin-interval-regression/58208/),
  [Software](https://github.com/tdhock/penaltyLearning),
  [Reproducible](https://gforge.inria.fr/scm/viewvc.php/pruned-dp/?root=breakpoints).

### 2012

- Hocking TD, Rigaill G. SegAnnot: an R package for fast segmentation
  of annotated piecewise constant signals,
  [Preprint hal-00759129](https://hal.inria.fr/hal-00759129). 
  [Software](https://r-forge.r-project.org/scm/?group_id=1541),
  [Reproducible](https://gforge.inria.fr/scm/viewvc.php/pruned-dp/SegAnnot/inst/doc/?root=breakpoints).
- Hocking TD. Learning algorithms and statistical software, with
  applications to bioinformatics. PhD thesis, Ecole Normale Superieure
  de Cachan,
  France. [tel-00906029](https://tel.archives-ouvertes.fr/tel-00906029/), 
  [Reproducible](https://rcdata.nau.edu/genomic-ml/public_html/papers/2012-11-20-PhD-thesis/HOCKING-phd-thesis.tgz).

### 2011

- Hocking TD, Joulin A, Bach F, Vert J-P. Clusterpath: an Algorithm
  for Clustering using Convex Fusion Penalties.  International
  Conference on Machine Learning
  (ICML), 2011. [pdf](http://www.icml-2011.org/papers/419_icmlpaper.pdf),
  [video](http://techtalks.tv/talks/clusterpath-an-algorithm-for-clustering-using-convex-fusion-penalties/54405/).
  [Software](https://r-forge.r-project.org/scm/?group_id=1090),
  [Reproducible](https://r-forge.r-project.org/scm/viewvc.php/tex/?root=clusterpath). Cited
  by Genevera Allen who showed interactive visualizations in her
  [Ihaka lecture](https://www.youtube.com/watch?v=2g-akN6q8aI).

### 2010

- Gautier M, Hocking TD, Foulley JL. A Bayesian outlier criterion to
  detect SNPs under selection in large data sets.  PloS ONE 5 (8),
  e11913
  (2010). [journal](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0011913).

### 2008

- Doyon Y, McCammon JM, Miller JC, Faraji F, Ngo C, Katibah GE, Amora
  R, Hocking TD, Zhang L, Rebar EJ, Gregory PD, Urnov FD, Amacher
  SL. Heritable targeted gene disruption in zebrafish using designed
  zinc-finger nucleases.  Nature biotechnology 26 (6), 702-70
  (2008). [pubmed](http://www.ncbi.nlm.nih.gov/pubmed/18500334).



