---
layout: page
permalink: /publications/
title: Publications
description: Peer-reviewed papers and preprints with links to code and data
---

- My pages on
  [Google Scholar](https://scholar.google.ca/citations?user=c83d8tgAAAAJ)
  and
  [Mathematics Genealogy Project](https://www.genealogy.math.ndsu.nodak.edu/id.php?id=224469).
- Peer review notice: in addition to peer-reviewed journals, I publish
  papers at machine learning conferences such as ICML and NeurIPS, which
  have double-blind peer reviews, and only accept about 20% of
  submitted papers.
- Reproducible Research notice: a "Reproducible" link for each paper below
  provides source code and data that I used to perform the analysis and make
  the figures/tables.
- Software notice: a "Software" link for each paper below provides a software
  package with a reference implementation of the algorithms described
  in our paper. See also my [Software](https://tdhock.github.io/software/) page.
- Please download my [TDH-refs.bib](/assets/TDH-refs.bib) BibTeX
  citation database, and use it to cite my papers. This is the file
  that was used to
  [generate](https://tdhock.github.io/blog/2024/auto-pubs-page/) the
  list of publications below.

<table style='width:30%;'>
 <thead>
  <tr>
   <th style="text-align:left;"> Type </th>
   <th style="text-align:right;"> count </th>
  </tr>
 </thead>
<tbody>
  <tr>
   <td style="text-align:left;"> journal paper </td>
   <td style="text-align:right;"> 34 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> in progress </td>
   <td style="text-align:right;"> 18 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> conference paper </td>
   <td style="text-align:right;"> 12 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> book chapter </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
  <tr>
   <td style="text-align:left;"> PHD thesis </td>
   <td style="text-align:right;"> 1 </td>
  </tr>
</tbody>
</table>



|Figure                                                                                                     |Published   |Authors                                                                                                                                                                                                                                                                                                                                                                                                                         |Title                                                                                                                     |Venue                                                                                                                                             |Links                                                                                                 |
|:----------------------------------------------------------------------------------------------------------|:-----------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------|:-----------------------------------------------------------------------------------------------------|
|<img src="/assets/img/publications/thumb/Gurney2024power.png" width="150" />                               |In progress |Gurney K, Aslam B, Dass P, Gawuc L, Hocking TD, Barber JJ, Kato A                                                                                                                                                                                                                                                                                                                                                               |Assessment of the Climate Trace global powerplant CO2 emissions                                                           |Accepted at Environmental Research Letters                                                                                                        |                                                                                                      |
|<img src="/assets/img/publications/thumb/Hocking2024soak.png" width="150" />                               |In progress |Bodine CS, Thibault G, Arellano PN, Shenkin AF, Lindly O, Hocking TD                                                                                                                                                                                                                                                                                                                                                            |SOAK: Same/Other/All K-fold cross-validation for estimating similarity of patterns in data subsets                        |In progress                                                                                                                                       |[Software](https://github.com/tdhock/mlr3resampling), [Reproducible](https://github.com/tdhock/cv-same-other-paper)|
|<img src="/assets/img/publications/thumb/Fowler2024line.png" width="150" />                                |In progress |Fowler J, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                            |Efficient line search for optimizing Area Under the ROC Curve in gradient descent                                         |In progress                                                                                                                                       |[Talk announcement for JSM'23 Toronto](https://ww2.aievolution.com/JSMAnnual/index.cfm?do=ev.viewEv&ev=2810), [Video of talk at Université Laval July 2023](https://www.youtube.com/watch?v=22ODpDTZ4VE), [Slides PDF](https://github.com/tdhock/max-generalized-auc/blob/master/HOCKING-slides-toronto.pdf), [source](https://github.com/tdhock/max-generalized-auc/#4-may-2023)|
|<img src="/assets/img/publications/thumb/Hocking2024binsegRcpp.png" width="150" />                         |In progress |Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                                      |Comparing binsegRcpp with other implementations of binary segmentation for changepoint detection                          |In progress                                                                                                                                       |[Reproducible](https://github.com/tdhock/binseg-model-selection), [Software](https://github.com/tdhock/binsegRcpp)|
|<img src="/assets/img/publications/thumb/Hocking2024finite.png" width="150" />                             |In progress |Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                                      |Finite Sample Complexity Analysis of Binary Segmentation                                                                  |In progress                                                                                                                                       |[Reproducible](https://github.com/tdhock/binseg-model-selection), [Software](https://github.com/tdhock/binsegRcpp)|
|<img src="/assets/img/publications/thumb/Hocking2024hmm.png" width="150" />                                |In progress |Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                                      |Teaching Hidden Markov Models Using Interactive Data Visualization                                                        |In progress                                                                                                                                       |[Reproducible](https://github.com/tdhock/2023-08-unsupervised-learning/blob/main/slides/07-hidden-markov-models.Rmd), [Slides](https://github.com/tdhock/2023-08-unsupervised-learning/blob/main/slides/07-hidden-markov-models.pdf)|
|<img src="/assets/img/publications/thumb/Hocking2024mlr3resampling.png" width="150" />                     |In progress |Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                                      |mlr3resampling: an R implementation of cross-validation for comparing models learned using different train subsets        |In progress                                                                                                                                       |[Software](https://github.com/tdhock/mlr3resampling)                                                  |
|<img src="/assets/img/publications/thumb/Hocking2024autism.png" width="150" />                             |In progress |Sutherland V, Hocking TD, Lindly O                                                                                                                                                                                                                                                                                                                                                                                              |Interpretable machine learning algorithms for understanding factors related to childhood autism                           |In progress                                                                                                                                       |[Reproducible](https://github.com/tdhock/2024-01-ml-for-autism)                                       |
|<img src="/assets/img/publications/thumb/Thibault2024forest.png" width="150" />                            |In progress |Thibault G, Morin-Bernard A, Sylvain J-D, Drolet G, Roussel J-R, Hocking TD, Achim A                                                                                                                                                                                                                                                                                                                                            |Spatial characterization of burn severity in a boreal forest using high-resolution satellite imagery                      |In progress                                                                                                                                       |                                                                                                      |
|<img src="/assets/img/publications/thumb/Truong2024circular.png" width="150" />                            |In progress |Truong C, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                            |Efficient change-point detection for multivariate circular data                                                           |In progress                                                                                                                                       |[Reproducible](https://github.com/tdhock/angular-change-paper), [Software](https://github.com/tdhock/geodesichange)|
|<img src="/assets/img/publications/thumb/Nguyen2024penalty.png" width="150" />                             |In progress |Nguyen TL, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                           |Penalty Learning for Optimal Partitioning through Deep Learning Techniques                                                |Under review at Computational Statistics                                                                                                          |[Preprint](https://arxiv.org/abs/2408.00856)                                                          |
|<img src="/assets/img/publications/thumb/Hocking2023functional.png" width="150" />                         |In progress |Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                                      |Why does functional pruning yield such fast algorithms for optimal changepoint detection?                                 |In progress                                                                                                                                       |Invited talk for TRIPODS seminar [video](https://arizona.hosted.panopto.com/Panopto/Pages/Viewer.aspx?id=4e87c8d0-96d2-40d1-808c-ad16014c6962), [slides](https://github.com/tdhock/functional-pruning-theory#readme), [IEEE NJACS](https://events.vtools.ieee.org/m/280515), [ASU West ML Day](https://newcollege.asu.edu/machine-learning-day-1)|
|<img src="/assets/img/publications/thumb/Rust2023pairs.png" width="150" />                                 |In progress |Rust KR, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                             |A Log-linear Gradient Descent Algorithm for Unbalanced Binary Classification using the All Pairs Squared Hinge Loss       |Preprint arXiv:2302.11062                                                                                                                         |[Preprint](https://arxiv.org/abs/2302.11062)                                                          |
|<img src="/assets/img/publications/thumb/Agyapong2023cv.png" width="150" />                                |In progress |Agyapong D, Propster JR, Marks J, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                    |Cross-Validation for Training and Testing Co-occurrence Network Inference Algorithms                                      |Under review at BMC Bioinformatics                                                                                                                |[Preprint](https://arxiv.org/abs/2309.15225)                                                          |
|<img src="/assets/img/publications/thumb/Hocking2017changepoint.png" width="150" />                        |In progress |Hocking TD, Killick R                                                                                                                                                                                                                                                                                                                                                                                                           |Introduction to optimal changepoint detection algorithms                                                                  |Tutorial at international useR 2017 conference, textbook in progress                                                                              |[Conference](https://www.user2017.brussels/uploads/TobyHocking.html), [Book](https://github.com/tdhock/changepoint-book), [Reproducible](https://github.com/tdhock/change-tutorial)|
|<img src="/assets/img/publications/thumb/Hocking2016interactive.png" width="150" />                        |In progress |Hocking TD, Ekstrøm CT                                                                                                                                                                                                                                                                                                                                                                                                          |Understanding and Creating Interactive Graphics                                                                           |Tutorial at international useR 2016 conference, textbook in progress                                                                              |[Conference](http://user2016.r-project.org/tutorials/13.html), [Manual](https://rcdata.nau.edu/genomic-ml/animint2-manual/Ch02-ggplot2.html), [Reproducible](https://github.com/tdhock/interactive-tutorial)|
|<img src="/assets/img/publications/thumb/Hocking2015breakpointError.png" width="150" />                    |In progress |Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                                      |A breakpoint detection error function for segmentation model selection and validation                                     |Preprint arXiv:1509.00368                                                                                                                         |[Preprint](https://arxiv.org/abs/1509.00368), [Software](http://r-forge.r-project.org/scm/?group_id=1540), [Reproducible](https://github.com/tdhock/breakpointError-orig)|
|<img src="/assets/img/publications/thumb/Venuto2014support.png" width="150" />                             |In progress |Venuto D, Hocking TD, Sphanurattana L, Sugiyama M                                                                                                                                                                                                                                                                                                                                                                               |Support vector comparison machines                                                                                        |Preprint arXiv:1401.8008                                                                                                                          |[Preprint](https://arxiv.org/abs/1401.8008), [Software](https://github.com/tdhock/rankSVMcompare), [Reproducible](https://github.com/tdhock/compare-paper)|
|<img src="/assets/img/publications/thumb/Kaufman2024functional.png" width="150" />                         |2024        |Kaufman JM, Stenberg AJ, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                             |Functional Labeled Optimal Partitioning                                                                                   |Journal of Computational and Graphical Statistics, DOI: 10 . 1080/10618600 . 2023 . 2293216                                                       |[DOI](https://doi.org/10.1080/10618600.2023.2293216), [Preprint](https://arxiv.org/abs/2210.02580), [Software](https://github.com/tdhock/FLOPART), [Reproducible](https://github.com/jkaufy/Flopart-Paper), [Preliminary code](https://github.com/tdhock/LabeledFPOP-paper)|
|<img src="/assets/img/publications/thumb/Bodine2024mapping.png" width="150" />                             |2024        |Bodine CS, Buscombe D, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                               |Automated River Substrate Mapping From Sonar Imagery With Machine Learning                                                |Journal of Geophysical Research: Machine Learning and Computation 1(3)                                                                            |[DOI](https://doi.org/10.1029/2024JH000135), [Preprint](https://eartharxiv.org/repository/view/6448/), [Software](https://github.com/CameronBodine/PINGMapper)|
|<img src="/assets/img/publications/thumb/Tao2024reply.png" width="150" />                                  |2024        |Tao F, Houlton BZ, Frey SD, Lehmann J, Manzoni S, Huang Y, Jiang L, Mishra U, Hungate BA, Schmidt MWI, Reichstein M, Carvalhais N, Ciais P, Wang Y-P, Ahrens B, Hugelius G, Hocking TD, Lu X, Shi Z, Viatkin K, Vargas R, Yigini Y, Omuto C, Malik AA, Peralta G, Cuevas-Corona R, Paolo LED, Luotto I, Liao C, Liang Y-S, Saynes VS, Huang X, Luo Y                                                                            |Reply to: Model uncertainty obscures major driver of soil carbon                                                          |Nature 627(8002)                                                                                                                                  |[Publisher](https://www.nature.com/articles/s41586-023-07000-9), [Preprint](https://eartharxiv.org/repository/view/6125/)|
|<img src="/assets/img/publications/thumb/Sweeney2023insect.png" width="150" />                             |2023        |Sweeney N, Xu C, Shaw JA, Hocking TD, Whitaker BM                                                                                                                                                                                                                                                                                                                                                                               |Insect Identification in Pulsed Lidar Images Using Changepoint Detection Algorithms                                       |2023 Intermountain Engineering, Technology and Computing (IETC)                                                                                   |[Publisher](https://ieeexplore.ieee.org/abstract/document/10152205), [Local PDF](/assets/papers/Sweeney2023insect.pdf)|
|<img src="/assets/img/publications/thumb/Hocking2023lopart.png" width="150" />                             |2023        |Hocking TD, Srivastava A                                                                                                                                                                                                                                                                                                                                                                                                        |Labeled optimal partitioning                                                                                              |Computational Statistics 38                                                                                                                       |[Publisher](https://rdcu.be/cQ8qM), [Preprint](https://arxiv.org/abs/2006.13967), [Video](https://www.youtube.com/watch?v=lm_6_33zOWc), [Software](https://github.com/tdhock/LOPART), [Reproducible](https://github.com/tdhock/LOPART-paper)|
|<img src="/assets/img/publications/thumb/Harshe2023exoskeleton.png" width="150" />                         |2023        |Harshe K, Williams JR, Hocking TD, Lerner ZF                                                                                                                                                                                                                                                                                                                                                                                    |Predicting Neuromuscular Engagement to Improve Gait Training With a Robotic Ankle Exoskeleton                             |IEEE Robotics and Automation Letters 8(8)                                                                                                         |[Publisher](https://ieeexplore.ieee.org/document/10172008), [Local PDF](/assets/papers/Harshe2023exoskeleton.pdf)|
|<img src="/assets/img/publications/thumb/Hillman2023jmlr.png" width="150" />                               |2023        |Hillman J, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                           |Optimizing ROC Curves with a Sort-Based Surrogate Loss for Binary Classification and Changepoint Detection                |Journal of Machine Learning Research 24(70)                                                                                                       |[Publisher](https://jmlr.org/papers/v24/21-0751.html), [Preprint](https://arxiv.org/abs/2107.01285), [Software](https://github.com/tdhock/aum), [Reproducible](https://github.com/tdhock/max-generalized-auc)|
|<img src="/assets/img/publications/thumb/Runge2023jss.png" width="150" />                                  |2023        |Runge V, Hocking TD, Romano G, Afghah F, Fearnhead P, Rigaill G                                                                                                                                                                                                                                                                                                                                                                 |gfpop: An R Package for Univariate Graph-Constrained Change-Point Detection                                               |Journal of Statistical Software 106(6)                                                                                                            |[Publisher](https://www.jstatsoft.org/article/view/v106i06), [Preprint](https://arxiv.org/abs/2002.03646), [Software](https://github.com/vrunge/gfpop), [GUI](https://github.com/julianstanley/gfpopgui), [Reproducible](https://github.com/vrunge/gfpop/blob/master/vignettes/applications.Rmd)|
|<img src="/assets/img/publications/thumb/Tao2023nature.png" width="150" />                                 |2023        |Tao F, Huang Y, Hungate BA, Manzoni S, Frey SD, Schmidt MWI, Reichstein M, Carvalhais N, Ciais P, Jiang L, Lehmann J, Mishra U, Hugelius G, Hocking TD, Lu X, Shi Z, Viatkin K, Vargas R, Yigini Y, Omuto C, Malik AA, Perualta G, Cuevas-Corona R, Paolo LED, Luotto I, Liao C, Liang Y-S, Saynes VS, Huang X, Luo Y                                                                                                           |Microbial carbon use efficiency promotes global soil carbon storage                                                       |Nature 618                                                                                                                                        |[Publisher](https://www.nature.com/articles/s41586-023-06042-3)                                       |
|<img src="/assets/img/publications/thumb/Barr2022classifying.png" width="150" />                           |2022        |Barr JR, Hocking TD, Morton G, Thatcher T, Shaw P                                                                                                                                                                                                                                                                                                                                                                               |Classifying Imbalanced Data with AUM Loss                                                                                 |2022 Fourth International Conference on Transdisciplinary AI (TransAI)                                                                            |[Publisher](https://ieeexplore.ieee.org/document/9951523), [Local PDF](/assets/papers/Barr2022classifying.pdf)|
|<img src="/assets/img/publications/thumb/Hocking2022interpretable.png" width="150" />                      |2022        |Hocking TD, Barr JR, Thatcher T                                                                                                                                                                                                                                                                                                                                                                                                 |Interpretable linear models for predicting security vulnerabilities in source code                                        |2022 Fourth International Conference on Transdisciplinary AI (TransAI)                                                                            |[Publisher](https://ieeexplore.ieee.org/document/9951610/), [Local PDF](/assets/papers/Hocking2022interpretable.pdf)|
|<img src="/assets/img/publications/thumb/Barr2022graph.png" width="150" />                                 |2022        |Barr JR, Shaw P, Abu-Khzam FN, Thatcher T, Hocking TD                                                                                                                                                                                                                                                                                                                                                                           |Graph embedding: A methodological survey                                                                                  |2022 fourth international conference on transdisciplinary AI (TransAI)                                                                            |[Publisher](https://ieeexplore.ieee.org/document/9951469/), [Local PDF](/assets/papers/Barr2022graph.pdf)|
|<img src="/assets/img/publications/thumb/Chaves2022chatbot.png" width="150" />                             |2022        |Chaves AP, Egbert J, Hocking T, Doerry E, Gerosa MA                                                                                                                                                                                                                                                                                                                                                                             |Chatbots Language Design: The Influence of Language Variation on User Experience with Tourist Assistant Chatbots          |ACM Trans. Comput.-Hum. Interact. 29(2)                                                                                                           |[Publisher](https://dl.acm.org/doi/10.1145/3487193), [Preprint](https://arxiv.org/abs/2101.11089), [Local PDF](/assets/papers/Chaves2022chatbot.pdf)|
|<img src="/assets/img/publications/thumb/Mihaljevic2022sparsemodr.png" width="150" />                      |2022        |Mihaljevic JR, Borkovec S, Ratnavale S, Hocking TD, Banister KE, Eppinger JE, Hepp C, Doerry E                                                                                                                                                                                                                                                                                                                                  |SPARSEMODr: Rapidly simulate spatially explicit and stochastic models of COVID-19 and other infectious diseases           |Biology Methods and Protocols 7(1)                                                                                                                |[Publisher](https://academic.oup.com/biomethods/advance-article/doi/10.1093/biomethods/bpac022/6680179), [Preprint](https://www.medrxiv.org/content/10.1101/2021.05.13.21256216v1), [Software](https://github.com/NAU-CCL/SPARSEMODr)|
|<img src="/assets/img/publications/thumb/Hocking22intro.png" width="150" />                                |2022        |Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                                      |Introduction to machine learning and neural networks                                                                      |Chapter in Land Carbon Cycle Modeling: Matrix Approach, Data Assimilation, and Ecological Forecasting, edited by Yiqi Luo, published by CRC Press |[Publisher](https://www.taylorfrancis.com/books/oa-edit/10.1201/9780429155659/land-carbon-cycle-modeling-yiqi-luo-benjamin-smith), [My chapter](https://raw.githubusercontent.com/tdhock/2020-yiqi-summer-school/master/HOCKING-chapter.pdf), [Reproducible](https://github.com/tdhock/2020-yiqi-summer-school#prepared-for-the-summer-school-4th-year-2021)|
|<img src="/assets/img/publications/thumb/Vargovich2022breakpoints.png" width="150" />                      |2022        |Vargovich J, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                         |Linear time dynamic programming for computing breakpoints in the regularization path of models selected from a finite set |Journal of Computational and Graphical Statistics 31(2)                                                                                           |[Publisher](https://amstat.tandfonline.com/doi/full/10.1080/10618600.2021.2000422), [Preprint](https://arxiv.org/abs/2003.02808), [Software](https://github.com/tdhock/penaltyLearning), [Reproducible](https://github.com/tdhock/changepoint-data-structure#source-code-for-figures-in-paper), [Local PDF](/assets/papers/Vargovich2022breakpoints.pdf)|
|<img src="/assets/img/publications/thumb/Barnwal2022jcgs.png" width="150" />                               |2022        |Barnwal A, Cho H, Hocking T                                                                                                                                                                                                                                                                                                                                                                                                     |Survival Regression with Accelerated Failure Time Model in XGBoost                                                        |Journal of Computational and Graphical Statistics 31(4)                                                                                           |[Publisher](https://www.tandfonline.com/doi/full/10.1080/10618600.2022.2067548), [Preprint](https://arxiv.org/abs/2006.04920), [Software](https://github.com/dmlc/xgboost), [Documentation](https://xgboost.readthedocs.io/en/latest/tutorials/aft_survival_analysis.html), [Video](https://www.youtube.com/watch?v=HuWRnzgGuIo), [Reproducible](https://github.com/avinashbarnwal/aftXgboostPaper/), [Local PDF](/assets/papers/Barnwal2022jcgs.pdf)|
|<img src="/assets/img/publications/thumb/Hocking2022jss.png" width="150" />                                |2022        |Hocking TD, Rigaill G, Fearnhead P, Bourque G                                                                                                                                                                                                                                                                                                                                                                                   |Generalized Functional Pruning Optimal Partitioning (GFPOP) for Constrained Changepoint Detection in Genomic Data         |Journal of Statistical Software 101(10)                                                                                                           |[Publisher](https://www.jstatsoft.org/article/view/v101i10), [Software](https://github.com/tdhock/PeakSegDisk), [Reproducible](https://github.com/tdhock/PeakSegFPOP-paper), [Slides](http://www.user2019.fr/static/pres/t257847.pdf), [Video](https://www.youtube.com/watch?v=XlC4WCqsbuI)|
|<img src="/assets/img/publications/thumb/Kolla2021fuzz.png" width="150" />                                 |2021        |Kolla AC, Groce A, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                   |Fuzz Testing the Compiled Code in R Packages                                                                              |2021 IEEE 32nd International Symposium on Software Reliability Engineering (ISSRE)                                                                |[Publisher](https://ieeexplore.ieee.org/document/9700272), [Software](https://github.com/akhikolla/RcppDeepState), [GitHub Action](https://github.com/FabrizioSandri/RcppDeepState-action), [Abstract](https://www.conftool.org/user2021/index.php?page=browseSessions&form_session=9#paperID123), [Video](https://www.youtube.com/watch?v=LXyruwSo2K0&t=245s), [Blog](https://akhikolla.github.io), [Results](https://akhikolla.github.io/packages-folders/), [Local PDF](/assets/papers/Kolla2021fuzz.pdf)|
|<img src="/assets/img/publications/thumb/Liehrmann2021chipseq.png" width="150" />                          |2021        |Liehrmann A, Rigaill G, Hocking TD                                                                                                                                                                                                                                                                                                                                                                                              |Increased peak detection accuracy in over-dispersed ChIP-seq data with supervised segmentation models                     |BMC Bioinformatics 22(323)                                                                                                                        |[Publisher](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-021-04221-5), [Reproducible](https://github.com/aLiehrmann/ChIP_SEQ_segmentation_paper), [Software](https://github.com/aliehrmann/crocs)|
|<img src="/assets/img/publications/thumb/Fotoohinasab2021greedy.png" width="150" />                        |2021        |Fotoohinasab A, Hocking T, Afghah F                                                                                                                                                                                                                                                                                                                                                                                             |A greedy graph search algorithm based on changepoint analysis for automatic QRS complex detection                         |Computers in Biology and Medicine 130                                                                                                             |[Publisher](https://www.sciencedirect.com/science/article/pii/S0010482521000020), [Preprint](https://arxiv.org/abs/2004.13558)|
|<img src="/assets/img/publications/thumb/Abraham2021gut.png" width="150" />                                |2021        |Abraham AJ, Prys-Jones TO, De Cuyper A, Ridenour C, Hempson GP, Hocking T, Clauss M, Doughty CE                                                                                                                                                                                                                                                                                                                                 |Improved estimation of gut passage time considerably affects trait-based dispersal models                                 |Functional Ecology 35(4)                                                                                                                          |[Publisher](https://besjournals.onlinelibrary.wiley.com/doi/10.1111/1365-2435.13726)                  |
|<img src="/assets/img/publications/thumb/Hocking2021reshaping.png" width="150" />                          |2021        |Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                                      |Wide-to-tall Data Reshaping Using Regular Expressions and the nc Package                                                  |The R Journal 13(1)                                                                                                                               |[Publisher](https://journal.r-project.org/archive/2021/RJ-2021-029/index.html), [Software](https://github.com/tdhock/nc), [Reproducible](https://github.com/tdhock/nc-article)|
|<img src="/assets/img/publications/thumb/Fotoohinasab2020segmentationEMBC.png" width="150" />              |2020        |Fotoohinasab A, Hocking T, Afghah F                                                                                                                                                                                                                                                                                                                                                                                             |A Graph-constrained Changepoint Detection Approach for ECG Segmentation                                                   |2020 42nd Annual International Conference of the IEEE Engineering in Medicine Biology Society (EMBC)                                              |[Publisher](https://ieeexplore.ieee.org/document/9175333), [Local PDF](/assets/papers/Fotoohinasab2020segmentationEMBC.pdf)|
|<img src="/assets/img/publications/thumb/Fotoohinasab2020automaticQRSdetectionAsilomar.png" width="150" /> |2020        |Fotoohinasab A, Hocking T, Afghah F                                                                                                                                                                                                                                                                                                                                                                                             |A Graph-Constrained Changepoint Learning Approach for Automatic QRS-Complex Detection                                     |2020 54th Asilomar Conference on Signals, Systems, and Computers                                                                                  |[Publisher](https://ieeexplore.ieee.org/document/9443307), [Abstract](https://www2.securecms.com/Asilomar2020/Papers/ViewPapers.asp?PaperNum=1323),  [Preprint](https://arxiv.org/abs/2004.13558), [Local PDF](/assets/papers/Fotoohinasab2020automaticQRSdetectionAsilomar.pdf)|
|<img src="/assets/img/publications/thumb/Hocking2020jmlr.png" width="150" />                               |2020        |Hocking TD, Rigaill G, Fearnhead P, Bourque G                                                                                                                                                                                                                                                                                                                                                                                   |Constrained Dynamic Programming and Supervised Penalty Learning Algorithms for Peak Detection in Genomic Data             |Journal of Machine Learning Research 21(87)                                                                                                       |[Publisher](http://jmlr.org/papers/v21/18-843.html), [Preprint](https://arxiv.org/abs/1703.03352), [Software](https://github.com/tdhock/PeakSegOptimal), [Reproducible](https://github.com/tdhock/PeakSegFPOP-paper)|
|<img src="/assets/img/publications/thumb/Hocking2020psb.png" width="150" />                                |2020        |Hocking TD, Bourque G                                                                                                                                                                                                                                                                                                                                                                                                           |Machine Learning Algorithms for Simultaneous Supervised Detection of Peaks in Multiple Samples and Cell Types             |Proc. Pacific Symposium on Biocomputing                                                                                                           |[Publisher](http://psb.stanford.edu/psb-online/proceedings/psb20/Hocking.pdf), [Software](https://github.com/tdhock/PeakSegJoint), [Preprint](https://arxiv.org/abs/1506.01286), [Reproducible](https://github.com/tdhock/PeakSegJoint-paper)|
|<img src="/assets/img/publications/thumb/Jewell2019biostatistics.png" width="150" />                       |2019        |Jewell SW, Hocking TD, Fearnhead P, Witten DM                                                                                                                                                                                                                                                                                                                                                                                   |Fast nonconvex deconvolution of calcium imaging data                                                                      |Biostatistics 21(4)                                                                                                                               |[Pubmed](https://www.ncbi.nlm.nih.gov/pubmed/30753436)                                                |
|<img src="/assets/img/publications/thumb/Sievert2019jcgs.png" width="150" />                               |2019        |Sievert C, VanderPlas S, Cai J, Ferris K, Khan FUF, Hocking TD                                                                                                                                                                                                                                                                                                                                                                  |Extending ggplot2 for Linked and Animated Web Graphics                                                                    |Journal of Computational and Graphical Statistics 28(2)                                                                                           |[Publisher](https://amstat.tandfonline.com/doi/full/10.1080/10618600.2018.1513367), [Software](https://github.com/tdhock/animint), [Reproducible](https://github.com/tdhock/animint-paper), [Interactive Figures](https://rcdata.nau.edu/genomic-ml/public_html/animint-paper-figures/), [Local PDF](/assets/papers/Sievert2019jcgs.pdf)|
|<img src="/assets/img/publications/thumb/Hocking2019regex.png" width="150" />                              |2019        |Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                                      |Comparing namedCapture with other R packages for regular expressions                                                      |The R Journal 11(2)                                                                                                                               |[Publisher](https://journal.r-project.org/archive/2019/RJ-2019-050/index.html), [Software](https://github.com/tdhock/namedCapture), [Reproducible](https://github.com/tdhock/namedCapture-article)|
|<img src="/assets/img/publications/thumb/Depuydt2018genomic.png" width="150" />                            |2018        |Depuydt P, Boeva V, Hocking TD, Cannoodt R, Ambros IM, Ambros PF, Asgharzadeh S, Attiyeh EF, Combaret Vé, Defferrari R, Fischer M, Hero B, Hogarty MD, Irwin MS, Koster J, Kreissman S, Ladenstein R, Lapouble E, Laureys Gè, London WB, Mazzocco K, Nakagawara A, Noguera R, Ohira M, Park JR, Pötschger U, Theissen J, Tonini GP, Valteau-Couanet D, Varesio L, Versteeg R, Speleman F, Maris JM, Schleiermacher G, Preter KD |Genomic amplifications and distal 6q loss: novel markers for poor survival in high-risk neuroblastoma patients            |JNCI: Journal of the National Cancer Institute 110(10)                                                                                            |[Publisher](https://academic.oup.com/jnci/advance-article/doi/10.1093/jnci/djy022/4921185)            |
|<img src="/assets/img/publications/thumb/Depuydt2018meta.png" width="150" />                               |2018        |Depuydt P, Koster J, Boeva V, Hocking TD, Speleman F, Schleiermacher G, De Preter K                                                                                                                                                                                                                                                                                                                                             |Meta-mining of copy number profiles of high-risk neuroblastoma tumors                                                     |Scientific data 5(1)                                                                                                                              |[Publisher](https://www.nature.com/articles/sdata2018240)                                             |
|<img src="/assets/img/publications/thumb/Alirezaie2018clinpred.png" width="150" />                         |2018        |Alirezaie N, Kernohan KD, Hartley T, Majewski J, Hocking TD                                                                                                                                                                                                                                                                                                                                                                     |ClinPred: Prediction Tool to Identify Disease-Relevant Nonsynonymous Single-Nucleotide Variants                           |The American Journal of Human Genetics 103(4)                                                                                                     |[DOI](https://doi.org/10.1016/j.ajhg.2018.08.005), [Software](http://github.com/tdhock/predict-clinically-pathogenic), [used in dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP)|
|<img src="/assets/img/publications/thumb/Drouin2017mmit.png" width="150" />                                |2017        |Drouin A, Hocking T, Laviolette F                                                                                                                                                                                                                                                                                                                                                                                               |Maximum Margin Interval Trees                                                                                             |Advances in Neural Information Processing Systems 30                                                                                              |[Publisher](http://papers.nips.cc/paper/7080-maximum-margin-interval-trees), [Software](https://github.com/aldro61/mmit), [Reproducible](https://github.com/tdhock/mmit-paper), [Preprint](https://arxiv.org/abs/1710.04234), [Video](https://www.youtube.com/watch?v=sNrMH9z1rb4)|
|<img src="/assets/img/publications/thumb/Hocking2017bioinfo.png" width="150" />                            |2017        |Hocking TD, Goerner-Potvin P, Morin A, Shao X, Pastinen T, Bourque G                                                                                                                                                                                                                                                                                                                                                            |Optimizing ChIP-seq peak detectors using visual labels and supervised machine learning                                    |Bioinformatics 33(4)                                                                                                                              |[Pubmed](https://www.ncbi.nlm.nih.gov/pubmed/27797775), [Software](https://github.com/tdhock/PeakError), [Reproducible](https://bitbucket.org/mugqic/chip-seq-paper), [Data](https://rcdata.nau.edu/genomic-ml/chip-seq-chunk-db/)|
|<img src="/assets/img/publications/thumb/Maidstone2017optimal.png" width="150" />                          |2017        |Maidstone R, Hocking T, Rigaill G, Fearnhead P                                                                                                                                                                                                                                                                                                                                                                                  |On optimal multiple changepoint algorithms for large data                                                                 |Statistics and Computing 27                                                                                                                       |[Publisher](https://link.springer.com/article/10.1007/s11222-016-9636-3), [Software](https://r-forge.r-project.org/R/?group_id=1851), [Reproducible](https://r-forge.r-project.org/scm/viewvc.php/benchmark/?root=opfp)|
|<img src="/assets/img/publications/thumb/Chicard2016cancer.png" width="150" />                             |2016        |Chicard M, Boyault S, Colmet Daage L, Richer W, Gentien D, Pierron G, Lapouble E, Bellini A, Clement N, Iacono I, Bréjon Sé, Carrere M, Reyes Cé, Hocking T, Bernard V, Peuchmaur M, Corradini Nè, Faure-Conter Cé, Coze C, Plantaz D, Defachelles AS, Thebaud E, Gambart M, Millot Féé, Valteau-Couanet D, Michon J, Puisieux A, Delattre O, Combaret Vé, Schleiermacher G                                                     |Genomic Copy Number Profiling Using Circulating Free Tumor DNA Highlights Heterogeneity in Neuroblastoma                  |Clinical Cancer Research 22(22)                                                                                                                   |[Publisher](http://clincancerres.aacrjournals.org/content/early/2016/11/03/1078-0432.CCR-16-0500)     |
|<img src="/assets/img/publications/thumb/Shimada2016leukemia.png" width="150" />                           |2016        |Shimada K, Shimada S, Sugimoto K, Nakatochi M, Suguro M, Hirakawa A, Hocking TD, Takeuchi I, Tokunaga T, Takagi Y, Sakamoto A, Aoki T, Naoe T, Nakamura S, Hayakawa F, Seto M, Tomita A, Kiyoi H                                                                                                                                                                                                                                |Development and analysis of patient-derived xenograft mouse models in intravascular large B-cell lymphoma                 |Leukemia 30(7)                                                                                                                                    |[Pubmed](https://www.ncbi.nlm.nih.gov/pubmed/27001523), [Local PDF](/assets/papers/Shimada2016leukemia.pdf)|
|<img src="/assets/img/publications/thumb/Hocking2015icml.png" width="150" />                               |2015        |Hocking TD, Rigaill G, Bourque G                                                                                                                                                                                                                                                                                                                                                                                                |PeakSeg: constrained optimal segmentation and supervised penalty learning for peak detection in count data                |Proc. 32nd ICML                                                                                                                                   |[Publisher](http://proceedings.mlr.press/v37/hocking15.html), [Video](http://videolectures.net/icml2015_hocking_count_data/?q=hocking), [Software](https://github.com/tdhock/PeakSegDP), [Reproducible](https://github.com/tdhock/PeakSeg-paper)|
|<img src="/assets/img/publications/thumb/Hocking2014bioinformatics.png" width="150" />                     |2014        |Hocking TD, Boeva V, Rigaill G, Schleiermacher G, Janoueix-Lerosey I, Delattre O, Richer W, Bourdeaut F, Suguro M, Seto M, Bach F, Vert JP                                                                                                                                                                                                                                                                                      |SegAnnDB: interactive Web-based genomic segmentation                                                                      |Bioinformatics 30(11)                                                                                                                             |[Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/24493034), [Software](https://github.com/tdhock/SegAnnDB), [Reproducible](https://gforge.inria.fr/scm/viewvc.php/breakpoints/webapp/applications-note/), [Preprint](https://hal.inria.fr/hal-00759129), [Package](https://r-forge.r-project.org/scm/?group_id=1541), [Reproducible](https://gforge.inria.fr/scm/viewvc.php/pruned-dp/SegAnnot/inst/doc/?root=breakpoints)|
|<img src="/assets/img/publications/thumb/Suguro2014cancer.png" width="150" />                              |2014        |Suguro M, Yoshida N, Umino A, Kato H, Tagawa H, Nakagawa M, Fukuhara N, Karnan S, Takeuchi I, Hocking TD, Arita K, Karube K, Tsuzuki S, Nakamura S, Kinoshita T, Seto M                                                                                                                                                                                                                                                         |Clonal heterogeneity of lymphoid malignancies correlates with poor prognosis                                              |Cancer Sci 105(7)                                                                                                                                 |[Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/24815991)                                                 |
|<img src="/assets/img/publications/thumb/Hocking2013bioinformatics.png" width="150" />                     |2013        |Hocking TD, Schleiermacher G, Janoueix-Lerosey I, Boeva V, Cappo J, Delattre O, Bach F, Vert J-P                                                                                                                                                                                                                                                                                                                                |Learning smoothing models of copy number profiles using breakpoint annotations                                            |BMC Bioinformatics 14(164)                                                                                                                        |[Publisher](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-14-164), [Software](https://cran.r-project.org/src/contrib/Archive/bams/), [Reproducible](https://rcdata.nau.edu/genomic-ml/public_html/papers/2012-01-27-Breakpoint-annotation-model-smoothing/HOCKING-breakpoint-annotation-model-smoothing.tgz)|
|<img src="/assets/img/publications/thumb/Hocking2013sustainable.png" width="150" />                        |2013        |Hocking TD, Wutzler T, Ponting K, Grosjean P                                                                                                                                                                                                                                                                                                                                                                                    |Sustainable, Extensible Documentation Generation using inlinedocs                                                         |Journal of Statistical Software 54                                                                                                                |[Publisher](https://www.jstatsoft.org/article/view/v054i06), [Software](https://cran.r-project.org/package=inlinedocs), [Reproducible](https://r-forge.r-project.org/scm/viewvc.php/tex/jss762/?root=inlinedocs)|
|<img src="/assets/img/publications/thumb/Hocking2013icml.png" width="150" />                               |2013        |Rigaill G, Hocking T, Vert J-P, Bach F                                                                                                                                                                                                                                                                                                                                                                                          |Learning sparse penalties for change-point detection using max margin interval regression                                 |Proc. 30th ICML                                                                                                                                   |[Publisher](http://proceedings.mlr.press/v28/hocking13.html), [Video](http://techtalks.tv/talks/learning-sparse-penalties-for-change-point-detection-using-max-margin-interval-regression/58208/), [Software](https://github.com/tdhock/penaltyLearning), [Reproducible](https://gforge.inria.fr/scm/viewvc.php/pruned-dp/?root=breakpoints)|
|<img src="/assets/img/publications/thumb/Hocking2012phd.png" width="150" />                                |2012        |Hocking TD                                                                                                                                                                                                                                                                                                                                                                                                                      |Learning algorithms and statistical software, with applications to bioinformatics                                         |PHD thesis, Ecole normale supérieure de Cachan                                                                                                    |[Publisher](https://tel.archives-ouvertes.fr/tel-00906029/), [Reproducible](https://rcdata.nau.edu/genomic-ml/public_html/papers/2012-11-20-PhD-thesis/HOCKING-phd-thesis.tgz)|
|<img src="/assets/img/publications/thumb/Hocking2011clusterpath.png" width="150" />                        |2011        |Hocking TD, Joulin A, Bach F, Vert J-P                                                                                                                                                                                                                                                                                                                                                                                          |Clusterpath an algorithm for clustering using convex fusion penalties                                                     |28th international conference on machine learning                                                                                                 |[Publisher](http://www.icml-2011.org/papers/419_icmlpaper.pdf), [Video](http://techtalks.tv/talks/clusterpath-an-algorithm-for-clustering-using-convex-fusion-penalties/54405/), [Software](https://r-forge.r-project.org/scm/?group_id=1090), [Reproducible](https://r-forge.r-project.org/scm/viewvc.php/tex/?root=clusterpath), [Cited in Ihaka lecture](https://www.youtube.com/watch?v=2g-akN6q8aI)|
|<img src="/assets/img/publications/thumb/Gautier2010bayesian.png" width="150" />                           |2010        |Gautier M, Hocking TD, Foulley J-L                                                                                                                                                                                                                                                                                                                                                                                              |A Bayesian outlier criterion to detect SNPs under selection in large data sets                                            |PLoS one 5(8)                                                                                                                                     |[Publisher](http://journals.plos.org/plosone/article?id=10.1371/journal.pone.0011913)                 |
|<img src="/assets/img/publications/thumb/Doyon2008heritable.png" width="150" />                            |2008        |Doyon Y, McCammon JM, Miller JC, Faraji F, Ngo C, Katibah GE, Amora R, Hocking TD, Zhang L, Rebar EJ, Gregory PD, Urnov FD, Amacher SL                                                                                                                                                                                                                                                                                          |Heritable targeted gene disruption in zebrafish using designed zinc-finger nucleases                                      |Nature biotechnology 26(6)                                                                                                                        |[Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/18500334)                                                 |


```
## Warning in system(paste("cd ../assets && rm -f TDH-refs.pdf TDH-refs.aux
## TDH-refs.bbl && pdflatex TDH-refs && bibtex TDH-refs && pdflatex TDH-refs &&
## pdflatex TDH-refs && evince TDH-refs.pdf &")): 'cd' not found
```
